src/MongoDB/BSON/functions.cpp

```diff
@@ -107,6 +113,7 @@ Variant HHVM_FUNCTION(MongoDBBsonToJson, const String &data)
 		bson_free(str);
 	} else {
 		throw MongoDriver::Utils::throwUnexpectedValueException("Could not read document from BSON reader");
+		bson_reader_destroy(reader);
```

Comment:
The same here - I guess doing the return Variant isn't really a *problem*, but it's dead code.